### PR TITLE
Copyedit Third Parties 2020

### PR DESCRIPTION
**Pull request by bazzadp**
Makes progress on #1432 and #901

Made the following changes:
- Added chart URLs, sheets Gids and SQL markup to all figures missing them.
-  Reviewed use of “third party” (noun) versus “third-party” (adjective) throughout text and charts.
- Change figcaptions to chart titles rather than notable takeaways which are held in text and descriptions.
- Changed headings in text and charts to sentence case
- Added some TODOs with questions for authors (will call these out separately below)
- Standard other edits (removed double spaces, added spaces before units, added links... etc.)
- Removed unedited flag

### CODE REVIEW

File: src/content/en/2020/third-parties.md [hunk 1]
```diff
@@ -20,35 +20,39 @@ featured_stat_2: 21.5%
...
+Third-party content is a critical component of most websites today. It powers everything: analytics, live chat, advertising, video sharing and more. Third-party content provides value by taking the heavy lifting off of site owners and allows them to focus on their core competencies.
 
 Many think of third-party content as being JavaScript-based, but the data shows that this is only true for 22% of requests. Third-party content comes in all forms, from images (37%) to audio (0.1%).
 
 In this chapter we will review the prevalence of third-party content and how this has changed since 2019. We will also review: the impact of third-party content on page weight (a good proxy for overall performance impact), scripts that load early in the page lifecycle, the impact of third-party content on browser CPU time, and how open third-parties are with their performance data.
 
 ## Definitions
 
+{# TODO (authors) We try to avoid having empty headers. Could we have an intro sentence or two here? #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:35]: @simonhearne can you add something here and remove this comment?

File: src/content/en/2020/third-parties.md [hunk 2]
```diff
@@ -68,76 +72,95 @@ _Note on CDNs: The CDN category here includes providers that provide resources o
...
+* Only the home pages are tested. Other pages may having difference third-party requirements.
 * Roughly 84% of all third-party domains by request volume have been identified and categorized. The remaining 16% fall into the "Other" category.
 
+Learn more about our [methodology](./methodology).
+
 ## Prevalence
 
 A good starting point for this analysis is to confirm the statement that third-party content is a critical component of most websites today. How many websites use third-party tags, and how many tags do they use?
 
+{# TODO Authors - is "tags" the right wording here? Should we just use "content"? #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:85]: @simonhearne WDYT?

File: src/content/en/2020/third-parties.md [hunk 3]
```diff
@@ -68,76 +72,95 @@ _Note on CDNs: The CDN category here includes providers that provide resources o
...
+
 These prevalence numbers show a slight increase on [the 2019 results](../2019/third-parties): 93.87% of pages in the desktop crawl had at least one third-party request, the number was slightly higher at 94.10% of pages in the mobile crawl. A brief look into the small number of pages with no third-party content revealed that many were adult sites, some were government domains and some were basic landing / holding pages with little content. It is fair to say that the vast majority of pages have at least one third-party.
 
+{# TODO authors - that's interesting that it's still increasing. I would have thought that potentially the prevalance of cookie banners preventing some third-party content loading by default might cause a slight decrease this year. Though I guess that's only few countries like Europe, and mostly concerned with Advertising/Tracking third-party content. Also cookie trackers themsleves are often implemented as third-party CMP. Still interesting none-the-less. Is it worth noting this? Maybe with link to privacy chapter. #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:103]: @simonhearne WDYT? I see you did cover this down below, and in conclusion, but still wonder if worth commenting on this here?

File: src/content/en/2020/third-parties.md [hunk 4]
```diff
@@ -68,76 +72,95 @@ _Note on CDNs: The CDN category here includes providers that provide resources o
...
+Further down the list at 2.43% is `ajax.googleapis.com`, Google's [Hosted Libraries project](https://developers.google.com/speed/libraries). Whilst loading a library such as jQuery from a hosted service is easy, the additional cost of a connection to a third-party domain may have a negative impact on performance. It is best to host all critical JavaScript and CSS on the root domain, if possible. There is also now no cache benefit to using a shared CDN resource, as all major browsers [partition caches by page](https://developers.google.com/web/updates/2020/10/http-cache-partitioning). Harry Roberts has written a detailed blog post on [how to host your own static assets](https://csswizardry.com/2019/05/self-host-your-static-assets/).
 
-## Page Weight Impact
+## Page weight impact
+
+{# TODO (authors) - again can we have an intro sentence or two to avoid an empty heading? Maybe with a link to the Page Weight chapter? #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:163]: @simonhearne

File: src/content/en/2020/third-parties.md [hunk 5]
```diff
@@ -164,89 +189,109 @@ On the other end of the spectrum, the categories CDN, Content and Hosting all re
...
-  height=371
+  height=371,
+  sheets_gid="1363055589",
+  sql_file="percent_of_third_party_cache.sql"
   )
 }}
 
 Breaking down by response type highlights some common offenders: xml and text responses are less likely to be cacheable. Surprisingly, less than two-thirds of images served by third-parties are cacheable. On further inspection, this is due to the use of tracking 'pixels' which are returned as non-cacheable zero-size image responses.
 
+{# TODO authors - I believe this could be explained more. Either here or in the paragraph above the chart. Tracking pixels are one of the few third-party requests that should be non-cacheable but this is neither mentioned above, nor really explained here as working as intended. #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:219]: @simonhearne WDYT?

File: src/content/en/2020/third-parties.md [hunk 6]
```diff
@@ -164,89 +189,109 @@ On the other end of the spectrum, the categories CDN, Content and Hosting all re
...
+  height=371,
+  sheets_gid="1056232541",
+  sql_file="distribution_of_3XX_response_body_size.sql"
   )
 }}
 
-The results show that the majority of 3XX responses are small: the 90th percentile is 420B, i.e. 90% of 3XX responses are 420 bytes or smaller. The 95th percentile is 6.5kB, the 99th is 36kB and the 99.9th is over 100kB! Whilst redirects may seem innocuous, 100kB is an unreasonable amount of bytes over the wire for a response that simply leads to another response.
+The results show that the majority of 3XX responses are small: the 90th percentile is 420 bytes, i.e. 90% of 3XX responses are 420 bytes or smaller. The 95th percentile is 6.5 kB, the 99th is 36 kB and the 99.9th is over 100 kB! Whilst redirects may seem innocuous, 100kB is an unreasonable amount of bytes over the wire for a response that simply leads to another response.
+
+{# TODO Authors/Analysts - is it weird to use kB (1000) instead of KB (1024)? #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:239]: @simonhearne WDYT?

File: src/content/en/2020/third-parties.md [hunk 7]
```diff
@@ -164,89 +189,109 @@ On the other end of the spectrum, the categories CDN, Content and Hosting all re
...
+This chart shows the probability density function of total page CPU time by the third-party categories present on each page. The median page is at 50 on the percentile axis. The data shows that all third-party categories follow a similar pattern, with the median page between 400 - 1,000 ms CPU time. The outlier here is advertising (in black): if a page has advertising tags it is much more likely to have high CPU usage during page load. The median page with advertising tags has a CPU load time of 1,500 ms, compared to 500 ms for pages without advertising. The high CPU load time at the lower percentiles indicates that even the fastest sites are impacted significantly by the presence of third-parties categorized as advertising.
 
 ## Other
+
+{# TODO Authors - again an intro sentence or two to avoid empty heading #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:280]: @simonhearne

File: src/content/en/2020/third-parties.md [hunk 8]
```diff
@@ -258,12 +303,14 @@ We know that adding arbitrary JavaScript to our sites introduces risks to both s
...
 
 While third-party content is critical to many websites, auditing the impact of each provider is critical to ensure that they do not significantly impact user experience, page weight or CPU utilization. There are often self-hosting options for the top contributors to third-party weight, this is especially worth considering as there is now no caching benefit to using shared assets:
 
 * Google Fonts allows [self-hosting](https://www.tunetheweb.com/blog/should-you-self-host-google-fonts/) the assets
 * JavaScript CDNs can be replaced with self-hosted assets
 * Experimentation scripts can be self-hosted, e.g. [Optimizely](https://help.optimizely.com/Set_Up_Optimizely/Optimizely_self-hosting_for_Akamai_users)
+
+{# TODO Authors. This kind of just ends. Should we have something else here to finish off the chapter? #}
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:316]: @simonhearne WDYT?

**Review by bazzadp — COMMENTED** [review]
Highlighting TODOs to be reviewed by @simonhearne

File: src/content/en/2020/third-parties.md (hunk 6 above)
Comment (simonhearne) [reply src/content/en/2020/third-parties.md:239]: Looking through the almanac it seems more common to use Kilobyte (kB) than Kibibyte (KiB). The kilobyte prefix is a lower case k according to ISO 80000:13

Comment (bazzadp) [reply src/content/en/2020/third-parties.md]: OK happy to leave, so will resolve this. At least it's clear what this is as opposed to the ambiguous KB!

**Review by simonhearne — APPROVED** [review]

File: src/content/en/2020/third-parties.md [hunk 9]
```diff
@@ -92,64 +101,73 @@ The chart below shows the distribution of pages by third-party count. The 10th p
...
+Further down the list at 2.43% is `ajax.googleapis.com`, Google's [Hosted Libraries project](https://developers.google.com/speed/libraries). Whilst loading a library such as jQuery from a hosted service is easy, the additional cost of a connection to a third-party domain may have a negative impact on performance. It is best to host all critical JavaScript and CSS on the root domain, if possible. There is also now no cache benefit to using a shared CDN resource, as all major browsers [partition caches by page](https://developers.google.com/web/updates/2020/10/http-cache-partitioning). Harry Roberts has written a detailed blog post on [how to host your own static assets](https://csswizardry.com/2019/05/self-host-your-static-assets/).
 
-## Page Weight Impact
+## Page weight impact
 
+Third-parties can have a significant impact on the weight of a page, measured as the number of bytes downloaded by the browser. The [page weight chapter](./page-weight) explores this in more detail, here we focus on the third-parties that have the greatest impact on page weight.
```
Comment (bazzadp) [inline src/content/en/2020/third-parties.md:157]:
```suggestion
Third-parties can have a significant impact on the weight of a page, measured as the number of bytes downloaded by the browser. The [Page Weight chapter](./page-weight) explores this in more detail, here we focus on the third-parties that have the greatest impact on page weight.
```

**Review by bazzadp — COMMENTED** [review]
Thanks @simonhearne . Will try to get this out today.

One minor nit on capitalising chapter name that I'll commit.

**Pull request merged**